src/lib/storage/chunk_encoder.hpp

```diff
@@ -32,6 +32,9 @@ using ChunkEncodingSpec = std::vector<SegmentEncodingSpec>;
 /**
  * @brief Interface for encoding chunks
  *
+ * TODO(all): make compress chunk thread-safe; if it gets called while another thread is inserting, things will likely
+ *            break.
```

Comment:
Update comment: This thingy does not guarantee thread-safety, the ChunkEncoderTask does

Reply:
Altered to `NOT thread-safe. In a multi-threaded context, the ChunkCompressionTask should invoke the ChunkEncoder.`